### PR TITLE
fix: persist session ID across hooks via file

### PR DESCRIPTION
**Pull request by marcodelpin**
## Summary

- Fix session ID mismatch between `session-register` and `file-claims` hooks
- Each hook runs as a separate Node.js process, so `process.env.COORDINATION_SESSION_ID` doesn't persist
- Now writes session ID to `~/.claude/.coordination-session-id` file for cross-process sharing

## Problem

The coordination layer hooks generate different session IDs because:
1. `session-register.ts` sets `process.env.COORDINATION_SESSION_ID`
2. `file-claims.ts` tries to read it, but gets `undefined` (different process)
3. Both fall back to `Date.now().toString(36)` → different IDs

This causes:
- All file claims to be orphaned (no matching session in `sessions` table)
- Cross-session conflict detection not working

## Solution

- `session-register.ts`: Write session ID to file after generating
- `file-claims.ts`: Read session ID from file before falling back to generation

## Test plan

- [x] Added 12 unit tests for session ID persistence (`session-id-persistence.test.ts`)
- [x] All tests pass: `npm test -- --filter "session-id-persistence"`
- [x] Rebuilt dist files included

## Breaking changes

None - backward compatible. Falls back to previous behavior if file doesn't exist.

---

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Cross-process session persistence for consistent coordination and recovery.
  * Hook commands now report awareness of peer sessions and provide contextual reminders.

* **Bug Fixes**
  * Robust handling of missing/unreadable session sources with deterministic fallbacks.
  * Trims stray whitespace and warns on existing file claims instead of failing.

* **Tests**
  * Comprehensive test suite covering persistence, cross-process consistency, fallbacks, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: .claude/hooks/src/file-claims.ts [hunk 1]
```diff
@@ -9,22 +9,60 @@
...
+  }
+
+  // Try reading from file (cross-process persistence)
+  const sessionFile = getSessionIdFile();
+  if (existsSync(sessionFile)) {
+    try {
+      const id = readFileSync(sessionFile, 'utf-8').trim();
+      if (id) return id;
+    } catch { /* ignore read errors */ }
+  }
```
[the remark this answers is not in the document]
Comment (marcodelpin) [reply .claude/hooks/src/file-claims.ts]: @coderabbitai resolve - Code refactored to shared/session-id.ts
